packages/editor/src/lib/editor/shapes/ShapeUtil.ts

```diff
@@ -266,6 +267,16 @@ export abstract class ShapeUtil<Shape extends TLUnknownShape = TLUnknownShape> {
 		return []
 	}
 
+	/**
+	 * Retrieves the shape's label, if it has one.
+	 *
+	 * @param shape - The shape.
+	 * @public
+	 */
+	getLabel(shape: Shape): Rectangle2d | null {
+		return null
+	}
```

Comment:
i'm not sure about adding this to the core API - imo this is quite an our-shapes-specific concept. Other shapes might have multiple labels, for example. 

Couple options here:
1. mark this as `@internal` for now so it's not a public API (although this means others can't get at this behaviour too)
2. `Geometry2d` already has an `isLabel` property. you can use this to e.g. have a group with a body shape, and a rectangle shape marked as a label. instead of `getLabel`, we could have a generic helper that iterates through a geometry and returns any parts marked with `isLabel` (i prefer this option)
3. it's probably fine we could just ship this anyway

Reply:
ah, i hadn't seen that Geometry2d had an `isLabel` property - that would be useful, and I agree about a generic helper to iterate through. i'll give it a go 👍 

i do think though that, inherently shapes probably are paired with text, i.e. the label concept. i think how Geometry2d is aware of labels is maybe indicative that it's just a useful pattern to have in our library. i guess i don't know if we want to be _too_ abstract about this.